Managers/ActivityMan.cpp

```diff
@@ -1,1852 +1,164 @@
+					g_AudioMan.QueueSilence(30);
+					g_AudioMan.QueueMusicStream("Base.rte/Music/Watts/Last Man.ogg");
+					g_AudioMan.QueueSilence(30);
+					g_AudioMan.QueueMusicStream("Base.rte/Music/dBSoundworks/cc2g.ogg");
+				}
+			}
 
-void Activity::SetupPlayers()
-{
-    // Clear out the settings, we'll set them up again
-    m_TeamCount = 0;
-    m_PlayerCount = 0;
-    // Don't clear a CPU team's active status though
-    for (int team = Activity::TEAM_1; team < MAXTEAMCOUNT; ++team)
-	{
-		m_TeamActive[team] = false;
+			m_Activity->Pause(pause);
+			g_ConsoleMan.PrintString("SYSTEM: Activity \"" + m_Activity->GetPresetName() + "\" was " + (pause ? "paused" : "resumed"));
+		} else {
+			g_ConsoleMan.PrintString("ERROR: No Activity to pause!");
+		}
 	}
 
-    int screenIndex = -1;
-    for (int player = Activity::PLAYER_1; player < MAXPLAYERCOUNT; ++player)
-    {
-        // Count all active players and teams
-        if (m_IsActive[player])
-        {
-            m_PlayerCount++;
-            if (!m_TeamActive[m_Team[player]])
-                m_TeamCount++;
-            // Also activate the team
-            m_TeamActive[m_Team[player]] = true;
-        }
-
-        // Calculate which screen each human player is using, based on how many non-human players there are before him
-        int screenIndex = -1;
-        if (m_IsActive[player] && m_IsHuman[player])
-        {
-            for (int p = Activity::PLAYER_1; p < MAXPLAYERCOUNT && p <= player; ++p)
-            {
-                if (m_IsActive[p] && m_IsHuman[p])
-                    screenIndex++;
-            }
-        }
-        m_PlayerScreen[player] = screenIndex;
-    }
-}
-
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Virtual method:  Start
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Officially starts this. Creates all the data etc necessary to start
-//                  the activity.
-
-int Activity::Start()
-{
-    m_ActivityState = RUNNING;
-    m_Paused = false;
-
-    // Reset the mousemoving so that it won't trap the mouse if the window isn't in focus (common after loading)
-	if (!g_FrameMan.IsInMultiplayerMode())
-	{
-		g_UInputMan.DisableMouseMoving(true);
-		g_UInputMan.DisableMouseMoving(false);
-
-		// Enable keys again
-		g_UInputMan.DisableKeys(false);
+	int ActivityMan::RestartActivity() {
+		// If we have a start activity set, then clone it and pass in. (have to clone, or will delete self in StartActivity)
+		if (m_StartActivity) {
+			return StartActivity(dynamic_cast<Activity *>(m_StartActivity->Clone()));
+		} else {
+			return StartActivity(m_DefaultActivityType, m_DefaultActivityName);
+		}
+		// Report that we had to start the default because there wasn't a specified start activity
```

Comment:
Shit like this is why I fucking hate stupid comments!!! This is really goddamn confusing actually, because I don't know if the comment is how it was intended to work, or the code is.

Basically, with the code as-is, the comment is 100% untrue because that return is never reached, as if we don't have an m_StartActivity set we return whether or not we successfully started the default activity.
I think you should rewrite this method very slightly to be
```
	int ActivityMan::RestartActivity() {
		if (m_StartActivity) {
			// Nice comment about why we're cloning here
			return StartActivity(dynamic_cast<Activity *>(m_StartActivity->Clone()));
		}
		return StartActivity(m_DefaultActivityType, m_DefaultActivityName);
	}
```